Tests/MongoSwiftTests/ServerSelectionTests.swift

```diff
@@ -46,16 +47,64 @@ private struct RTTCalculationTestFile: Decodable {
     }
 }
 
+private struct SelectionWithinLatencyWindowTest: Decodable {
+    let topologyDescription: TopologyDescription
+    let mockedTopologyState: [TestServer]
+    let iterations: Int
+    let outcome: Outcome
+
+    fileprivate struct TestServer: Decodable {
+        let address: ServerAddress
+        let operationCount: Int
+
+        enum CodingKeys: String, CodingKey {
+            case address, operationCount = "operation_count"
+        }
+    }
+
+    fileprivate struct Outcome: Decodable {
+        let tolerance: Double
+        let expectedFrequencies: [ServerAddress: Double]
+
+        enum CodingKeys: String, CodingKey {
+            case tolerance, expectedFrequencies = "expected_frequencies"
+        }
+
+        internal init(from decoder: Decoder) throws {
+            let values = try decoder.container(keyedBy: CodingKeys.self)
+            self.tolerance = try values.decode(Double.self, forKey: .tolerance)
+            let expectedFrequenciesDocument = try values.decode(BSONDocument.self, forKey: .expectedFrequencies)
```

Comment:
nope, decoding failed because whole-number frequencies are specified as integers in the spec tests, but the defined test type expects doubles

Reply:
ah, I think we are also probably running into the issue described here with using a non-string, non-integer type for the dictionary keys: [SE-0320](https://github.com/apple/swift-evolution/blob/main/proposals/0320-codingkeyrepresentable.md) 

basically, even though it seems like a `[ServerAddress: Double]` should be decodable from a JSON object mapping `[String: Number]` since a `ServerAddress` can be initialized from a `String`, the `Dictionary` `Codable` implementation assumes that since `ServerAddress` is not a `String` that the corresponding JSON data is stored in an array of alternating key-value pairs, rather than an object